pkg/machineproviders/providers/openshift/machine/v1beta1/mapping.go

```diff
@@ -51,21 +46,21 @@ var (
 // to by external code to create new Machines in the same failure domain. It should start with a basic mapping and
 // then use existing Machine information to map failure domains, if possible, so that the Machine names match the
 // index of the failure domain in which they currently reside.
-func mapMachineIndexesToFailureDomains(ctx context.Context, logger logr.Logger, cl client.Client, cpms *machinev1.ControlPlaneMachineSet, failureDomains []failuredomain.FailureDomain) (map[int32]failuredomain.FailureDomain, error) {
+func mapMachineIndexesToFailureDomains(logger logr.Logger, machineList []machinev1beta1.Machine, replicas int32, failureDomains []failuredomain.FailureDomain) (map[int32]failuredomain.FailureDomain, error) {
```

Comment:
```suggestion
func mapMachineIndexesToFailureDomains(logger logr.Logger, machines []machinev1beta1.Machine, replicas int32, failureDomains []failuredomain.FailureDomain) (map[int32]failuredomain.FailureDomain, error) {
```
Confusing with `machineList` type.